whelktool/scripts/cleanups/2020/09/lxl-3152-linkOnlyMainEntity.groovy

```diff
@@ -0,0 +1,51 @@
+import whelk.Whelk
+
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+PrintWriter linkChanges = getReportWriter("link-changes")
+
+String where = "collection <> 'definitions'"
+
+selectBySqlWhere(where) { data ->
+    boolean changed = traverse(data.graph, data.doc.shortId, data.whelk, linkChanges)
+
+    if (changed) {
+        scheduledForUpdating.println("${data.doc.getURI()}")
+        data.scheduleSave(onError: { e ->
+            failedUpdating.println("Failed to update ${data.doc.shortId} due to: $e")
+        })
+    }
+}
+
+boolean traverse(Object node, String idForLogging, Whelk whelk, PrintWriter linkChanges) {
+    boolean changed = false
+
+    if (node instanceof Map) {
+        Map map = node
+
+        // Correct references are all id.kb.se or ends with #it.
```

Comment:
Code is correct but comment is wrong, could also be an external link (e.g. http://id.loc.gov/authorities/label/Wolves)

Reply:
:+1: